google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -103,14 +113,40 @@ public void addSupport(@NotNull Module module, @NotNull ModifiableRootModel root
       AppEngineFlexibleFacet facet = FacetManager.getInstance(module).addFacet(
           facetType, facetType.getPresentableName(), null /* underlyingFacet */);
 
-      // Allows suggesting app.yaml and Dockerfile locations in facet and deployment UIs.
       VirtualFile[] contentRoots = rootModel.getContentRoots();
       AppEngineProjectService appEngineProjectService = AppEngineProjectService.getInstance();
+
       if (contentRoots.length > 0) {
-        facet.getConfiguration().setAppYamlPath(
+        Path appYamlPath = Paths.get(
             appEngineProjectService.getDefaultAppYamlPath(contentRoots[0].getPath()));
+
+        // Allows suggesting app.yaml and Dockerfile locations in facet and deployment UIs.
+        facet.getConfiguration().setAppYamlPath(appYamlPath.toString());
         facet.getConfiguration().setDockerfilePath(
             appEngineProjectService.getDefaultDockerfilePath(contentRoots[0].getPath()));
+
+        if (generateConfigurationFilesCheckBox.isSelected()) {
+          if (Files.exists(appYamlPath)) {
+            try {
+              Files.move(appYamlPath,
+                  appYamlPath.getParent().resolve(
+                      appYamlPath.getFileName().toString() + ".bak"));
+            } catch (IOException e) {
+              // Do nothing for now.
```

Comment:
I was thinking we should maybe log something, not sure about aborting. We'd need to make sure that add project is transactional, and undo everything if this fails here. Realistically, I also don't see this happening very often, if ever?